### PR TITLE
Completed task with testing

### PR DESCRIPTION
**Pull request by progdev8201**

### CODE REVIEW

File: src/main/java/cal/internshipmanager/controller/InternshipApplicationController.java [hunk 1]
```diff
@@ -41,4 +43,15 @@ public void create(@Valid @RequestBody InternshipApplicationCreationRequest requ
     public InternshipApplicationListResponse internshipApplications(@PathVariable UUID userUniqueId) {
         return internshipApplicationService.internshipApplications(userUniqueId);
     }
+
+    @GetMapping()
+    public InternshipApplicationListResponse findByStatus(@RequestBody @Valid InternshipApplicationFindByStatusRequest request) {
```
Comment (MathieuDonofrio) [inline src/main/java/cal/internshipmanager/controller/InternshipApplicationController.java]: Use different service for every type like in InternshipOffer:
    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
    @GetMapping("pending-approval")
    public InternshipOfferListResponse pendingApproval() {
        return internshipOfferService.pendingApproval();
    }

    @PreAuthorize("hasAuthority('ADMINISTRATOR')")
    @GetMapping("rejected")
    public InternshipOfferListResponse rejected() {
        return internshipOfferService.rejected();
    }

    @PreAuthorize("hasAuthority('STUDENT') or hasAuthority('ADMINISTRATOR')")
    @GetMapping("approved")
    public InternshipOfferListResponse approved() {
        return internshipOfferService.approved();
    }

File: src/main/java/cal/internshipmanager/request/InternshipApplicationEditRequest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.UUID;
+
+@Data
+public class InternshipApplicationEditRequest implements Serializable {
+
+    @NotNull(message = "Application unique id is mandatory")
+    @ExistingInternshipApplication
+    private UUID applicationId;
+
+    @NotNull
```
Comment (MathieuDonofrio) [inline src/main/java/cal/internshipmanager/request/InternshipApplicationEditRequest.java:19]: Add message

File: src/main/java/cal/internshipmanager/request/InternshipApplicationFindByStatusRequest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import cal.internshipmanager.model.InternshipApplication;
+import cal.internshipmanager.validator.ExistingInternshipApplication;
+import lombok.Data;
+
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.UUID;
+
+@Data
+public class InternshipApplicationFindByStatusRequest implements Serializable {
```
Comment (MathieuDonofrio) [inline src/main/java/cal/internshipmanager/request/InternshipApplicationFindByStatusRequest.java]: remove

File: src/main/java/cal/internshipmanager/service/InternshipApplicationService.java [hunk 4]
```diff
@@ -79,8 +80,8 @@ public InternshipApplicationListResponse internshipApplications(@NotNull UUID us
         return response;
     }
 
-    public InternshipApplicationListResponse findByStatus(@NotBlank InternshipApplication.Status status) {
-        List<InternshipApplication> allApplications = internshipApplicationRepository.findAllByStatus();
+    public InternshipApplicationListResponse findByStatus(@Valid InternshipApplicationFindByStatusRequest request) {
```
Comment (MathieuDonofrio) [inline src/main/java/cal/internshipmanager/service/InternshipApplicationService.java]: change according to comments from controller

File: src/main/java/cal/internshipmanager/service/UserService.java [hunk 5]
```diff
@@ -35,7 +37,7 @@ public UserService(UserRepository userRepository) {
     //
 
     public UserListReponse students() {
-
+        //todo why is it called internshipoffers?
         List<User> internshipOffers = userRepository.findAllByType("STUDENT");
```
Comment (MathieuDonofrio) [inline src/main/java/cal/internshipmanager/service/UserService.java]: Just change name to users

**Review by MathieuDonofrio — CHANGES_REQUESTED** [review]

File: src/main/java/cal/internshipmanager/controller/InternshipApplicationController.java [hunk 6]
```diff
@@ -1,6 +1,8 @@
 package cal.internshipmanager.controller;
```
Comment (MathieuDonofrio) [inline src/main/java/cal/internshipmanager/controller/InternshipApplicationController.java:2]: Add default status when creating application

**Review by MathieuDonofrio — CHANGES_REQUESTED** [review]

File: src/main/java/cal/internshipmanager/request/InternshipApplicationEditRequest.java (hunk 2 above)
Comment (progdev8201) [reply src/main/java/cal/internshipmanager/request/InternshipApplicationEditRequest.java:19]: The message is already set in default in the validation class

File: src/main/java/cal/internshipmanager/controller/InternshipApplicationController.java (hunk 6 above)
Comment (progdev8201) [reply src/main/java/cal/internshipmanager/controller/InternshipApplicationController.java:2]: I added the status pending approval when its created

File: src/main/java/cal/internshipmanager/service/UserService.java (hunk 5 above)
Comment (progdev8201) [reply src/main/java/cal/internshipmanager/service/UserService.java]: done

**Review by MathieuDonofrio — APPROVED** [review]

**Pull request merged**